File: Basics of C Programming/perfect_number.c
#include <stdio.h>

int main(){
    int a,sum=0;
    printf("Enter the number:");
    scanf("%d",&a);
    int b=(a/2)+1;
    for(int i=1;i<b;i++){
        if (a%i==0){
            sum+=i;
        }
    }
    if (sum==a){
        printf("The %d is a perfect number",a);
    }else{
        printf("The %d is not a perfect number",a);
    }
    return 0;
}

File: python/python assignment 10 qus/perfect number.py
a=int(input("Enter the number:"))
sum=0
if a<0:
    print("The number is a negative number")
elif a==0:
    print("The number is zero")
else:
    c=(a//2)+1
    for i in range(1,c):
        if a%i==0:
            sum+=i
    if sum==a:
        print("It is a perfect number:")
    else:
        print("It is not a perfect number:")